tests/lib/keys/KeyManager.test.ts

```diff
@@ -8,69 +8,78 @@ describe('KeyManager class', () => {
   let tempDir: string
   let km: KeyManager
 
-  beforeAll(async done => {
+  beforeAll(async () => {
     // Define temp directory
     tempDir = fs.mkdtempSync(`${os.tmpdir}/pktest${randomString()}`)
 
     // Create keyManager
     km = new KeyManager(tempDir, fs)
     await km.generateKeyPair('John Smith', 'john.smith@email.com', 'passphrase', 1024, true)
-
-    done()
   })
 
   afterAll(() => {
     fs.rmdirSync(tempDir, { recursive: true })
   })
 
-  test('can create keypairs', async done => {
+  test('can create keypairs', async () => {
     // Create private keys (async)
     expect(km.generateKeyPair('John Smith', 'john.smith@gmail.com', 'passphrase', 1024)).resolves.not.toThrow()
-
-    done()
   })
 
-  test('can create symmetric keys', async done => {
+  test('can create symmetric keys', async () => {
     const generatedKey = await km.generateKey('new-key', 'passphrase', true)
 
     const retreivedKey = km.getKey('new-key')
 
     expect(retreivedKey).toEqual(generatedKey)
-
-    done()
   })
 
-  test('can load an identity from a public key', async done => {
+  test('can load an identity from a public key', async () => {
 
     const keypair = await km.generateKeyPair('John Smith', 'john@email.com', 'passphrase', 1024)
 
     const identity = await km.getIdentityFromPublicKey(Buffer.from(keypair.public!))
 
     expect(identity).not.toEqual(undefined)
-
-    done()
   })
 
-  test('can load an identity from a private key', async done => {
+  test('can load an identity from a private key', async () => {
 
     const keypair = await km.generateKeyPair('John Smith', 'john@email.com', 'passphrase', 1024)
 
     const identity = await km.getIdentityFromPrivateKey(Buffer.from(keypair.private!), 'passphrase')
 
     expect(identity).not.toEqual(undefined)
-
-    done()
   })
 
-  test('can sign and verify data', async done => {
+  test('can sign and verify data', async () => {
     const originalData = Buffer.from('I am to be signed')
     const signedData = await km.signData(originalData)
-    const isVerified = await km.verifyData(signedData)
-    expect(isVerified).toEqual(true)
-    done()
+    const verifiedData = await km.verifyData(signedData)
+    expect(verifiedData).toEqual(originalData)
+	})
+
+  test('can sign and verify data', async () => {
+    const originalData = Buffer.from('I am to be signed')
+    const detachedSignature = await km.signData(originalData)
+    const isVerified = await km.verifyData(detachedSignature)
+    expect(isVerified).toEqual(originalData)
+  })
+
+  test('can box and unbox a Uint8Array', async () => {
```

Comment:
What does boxing and unboxing mean?

Reply:
so boxing and unboxing are terms that keybase use to denote encrypting & signing or decrypting & verifying respectively. Also their methods are labelled this way as well.